core/services/registrysyncer/orm.go

```diff
@@ -0,0 +1,51 @@
+package registrysyncer
+
+import (
+	"context"
+	"crypto/sha256"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/sqlutil"
+
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+)
+
+type syncerORM struct {
+	ds   sqlutil.DataSource
+	lggr logger.Logger
+}
+
+func newORM(ds sqlutil.DataSource, lggr logger.Logger) syncerORM {
+	namedLogger := lggr.Named("RegistrySyncerORM")
+	return syncerORM{
+		ds:   ds,
+		lggr: namedLogger,
+	}
+}
+
+func (orm syncerORM) addState(ctx context.Context, state State) error {
+	stateJSON, err := state.MarshalJSON()
+	if err != nil {
+		return err
+	}
+	hash := sha256.Sum256(stateJSON)
+	_, err = orm.ds.ExecContext(
+		ctx,
+		`INSERT INTO registry_syncer_states (data, data_hash) VALUES ($1, $2) ON CONFLICT (data_hash) DO NOTHING`,
```

Comment:
Do we only add state and never prune it? I don't think there's a need to store all historical data. I'd either simplify to just storing single latest state or to store a limited number of updates and prune oldest ones each time.